src/ark/analysis/neighborhood_analysis.py

```diff
@@ -509,3 +510,64 @@ def compute_mixing_score(fov_neighbors_mat, fov, target_cells, reference_cells,
         mixing_score = reference_target / (target_target + reference_reference)
 
     return mixing_score
+
+
+def shannon_diversity(proportions):
+    """ Calculates the shannon diversity index for the provided proportions of a community
+    Args:
+        proportions (np.array): the proportions of each individual group
+
+    Returns:
+        float:
+            the diversity of neighborhood
+    """
+
+    proportions = [prop for prop in proportions if prop > 0]
```

Comment:
`numpy` indexing allows you to do:

```
prop_index = proportions > 0
return -np.sum(proportions[prop_index] * np.log2(proportions[prop_index]))
```

instead.

Reply:
Also, out of curiosity, is there a reason we're using base 2 log instead of the natural log?

Reply:
I think it should be natural log, but for now it's just based on the example code in #982.